pkg/promise/promise_test.go

```diff
@@ -0,0 +1,28 @@
+package promise
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestGo(t *testing.T) {
+	errCh := Go(functionWithError)
+	er := <-errCh
+	require.NotNil(t, er.Error())
+	require.EqualValues(t, "Error Occured", er.Error())
+	t.Log("GO promise with channel value 'Error' executed successfully")
```

Comment:
Maybe this is a good situation for using subtests? https://blog.golang.org/subtests